File: explicit-python/explicit/api/IConversionBinding.py
from abc import ABC, abstractmethod
from typing import Callable, List, Any


class IConversionBinding(ABC):

    @abstractmethod
    def register_variable(self, name: str, value):
        pass

    @abstractmethod
    def register_function(self, name:str, function: Callable[[List[Any]], Any]):
        pass

    @abstractmethod
    def get_variable(self, name: str) -> Any:
        pass

    @abstractmethod
    def get_function(self, name: str) -> Any:
        pass

File: explicit-java/src/main/kotlin/explicit/api/IConversionBinding.kt
package explicit.api

interface IConversionBinding {
    fun registerVariable(name:String, value:Any)
    fun registerFunction(name:String, function:(args:List<Any>) -> Any)

    fun getVariable(name:String):Any?
    fun getFunction(name:String):(List<Any>) -> Any
}
